.github/workflows/main.yml

```diff
@@ -375,6 +375,7 @@ jobs:
         path: benchmarks
 
   dist_linux_x86_64_musl:
+    if: github.ref == 'refs/heads/main'
```

Comment:
In practice nobody uses the nightly artifacts I believe, we could even do this only on a tag push, like deployment.

In any case, it doesn't hurt that we check if its not broken.

Reply:
Yeah I figured the time it takes on pushes to main is less critical than iterating on PRs.